### PR TITLE
Add polynomial_kernel wrapper (#2312)

### PR DESCRIPTION
**Pull request by SnigdhaSarkar16**
This PR adds a wrapper for the polynomial kernel in the scikit-learn-intelex extension. 

- Implements the polynomial kernel function similar to linear, rbf, and sigmoid kernels.
- Supports gamma, coef0, degree, and optional second array Y.
- Uses _compute_kernel with the backend.polynomial_kernel.
- Includes input validation via _check_inputs.

Fixes / relates to issue #2312

### CODE REVIEW

**Review by icfaust — CHANGES_REQUESTED** [review]
Thank you for your contribution! The Polynomial kernel already exists as `poly_kernel` in the same file. If we are to add a sklearnex interface, would need to be in the `sklearnex` module.

**Comment by SnigdhaSarkar16** [comment]
Thank you! I understand that the polynomial kernel already exists as poly_kernel. I’ll work on adding a sklearnex interface in the appropriate module instead.
